MariaDB/server
# MDEV-32637 Implement native UUID7 function

- [x] *The Jira issue number for this PR is: MDEV-32637*

## Description
MariaDB provides the UUID method that generates UUID v1, however, UUID v1 generates non-linear UUIDS that will spread data across different linear partitions based on key or hash.

The new UUID7 (https://uuid7.com/) will guarantee that continuous records are time-sorted and a good collision resistance.

The main advantage of the UUID7 function is that it will allow MariaDB to avoid the limitation of using a unique combined primary key for time series records, where a unique ID and a date are required to create key/hash partitions. With UUID7 is possible to only use the key as ID and time reference.

## Release Notes

Adds the function:
`UUIDv7()`: returns a UUIDv7

## How can this PR be tested?

This PR adds automatic tests that partially test the new functions. 

I am not aware of an easy way to change the time retrieved by `my_hrtime()`, hence, I could not write automatic tests to verify that the UUIDs generated are monotonic.
I am open to suggestions on how to accomplish that.

## Basing the PR against the correct MariaDB version

## PR quality check
